matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/workmanager/DefaultWorkManagerConfig.kt

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2023 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.matrix.android.sdk.internal.session.workmanager
+
+import org.matrix.android.sdk.api.auth.data.Credentials
+import org.matrix.android.sdk.api.extensions.orFalse
+import org.matrix.android.sdk.internal.session.homeserver.HomeServerCapabilitiesDataSource
+import javax.inject.Inject
+
+internal class DefaultWorkManagerConfig @Inject constructor(
+        private val credentials: Credentials,
+        private val homeServerCapabilitiesDataSource: HomeServerCapabilitiesDataSource,
+) : WorkManagerConfig {
+    override fun withNetworkConstraint(): Boolean {
```

Comment:
This algorithm is fine for me. I just want to highlight that if the NetworkConstraint is disabled at the login period, it would not be possible to enable it anymore via the Wellknown response.
The opposite (disable the NetworkConstraint if it was not already disabled at the login period) is supported , which is good to have.

Reply:
You're right, it's better to make this field nullable. Let me push another commit.

Reply:
64a7de53261ad6ba6c0bdee6c8c7346aebc3b413 let me know what you think @giomfo.

**Note**: I did not add a new db migration for this, it's fine since I have changed the migration on the same branch (so it will not break `develop`), but you may have to clear your data.

Reply:
thx, FYI here is my suggested algo since HomeServerCapabilitiesEntityFields.DISABLE_NETWORK_CONSTRAINT is nullable
```
override fun withNetworkConstraint(): Boolean {
        val disableNetworkConstraint = homeServerCapabilitiesDataSource.getHomeServerCapabilities()?.disableNetworkConstraint
        return if (disableNetworkConstraint != null) {
            // Boolean `io.element.disable_network_constraint` explicitly set in the .well-known file
            disableNetworkConstraint.not()
        }
        else if (credentials.discoveryInformation?.disableNetworkConstraint == true) {
            // Boolean `io.element.disable_network_constraint` explicitly set to `true` in the login response
            false
        }
        else {
            // Default, use the Network constraint
            true
        }
    }
```
I ran a test with this suggested algo, I don't understand why disableNetworkConstraint is always null even when a well-known response has been received